c7n/actions.py

```diff
@@ -257,6 +257,11 @@ def get_groups(self, resources, metadata_key=None):
                     rgroups = [g[metadata_key] for g in r['VpcSecurityGroups']]
                 else:
                     rgroups = [g for g in r['VpcSecurityGroups']]
+            elif r.get('VPCOptions', {}).get('SecurityGroupIds', []):
```

Comment:
this block needs at a minimum a comment on which resource it applies to, and frankly at this point i think this whole conditional needs to be split out to a method and made a subclassing responsibility.